obspy/clients/fdsn/mass_downloader/__init__.py

```diff
@@ -0,0 +1,503 @@
+
+
+Usage
+-----
+
+Using the download helpers requires the definition of three separate things,
+all of which are detailed in the following paragraphs.
+
+1. **Data Selection:** The data to be downloaded can be defined by enforcing
+   geographical or temporal constraints and a couple of other options.
+2. **Storage Options:** Choosing where the final MiniSEED and StationXML files
+   should be stored.
+3. **Start the Download:** Choose from which provider(s) to download and then
+   launch the downloading process.
+
+
+Step 1: Data Selection
+~~~~~~~~~~~~~~~~~~~~~~
+
+Data set selection serves the purpose to limit the data to be downloaded to
+data useful for the purpose at hand. It is handled by two objects:
+subclasses of the  :class:`~obspy.clients.fdsn.mass_downloader.domain.Domain`
+object and the
+:class:`~obspy.clients.fdsn.mass_downloader.restrictions.Restrictions` class.
+
+The :class:`~obspy.clients.fdsn.mass_downloader.domain` module currently
+defines three different domain types used to limit the geographical extent of
+the queried data:
+:class:`~obspy.clients.fdsn.mass_downloader.domain.RectangularDomain`,
+:class:`~obspy.clients.fdsn.mass_downloader.domain.CircularDomain`, and
+:class:`~obspy.clients.fdsn.mass_downloader.domain.GlobalDomain`. Subclassing
+:class:`~obspy.clients.fdsn.mass_downloader.domain.Domain` enables the
+construction of arbitrary complex domains. Please see the
+:class:`~obspy.clients.fdsn.mass_downloader.domain` module for more details.
+Instances of these classes will later be passed to the function sparking the
+downloading process. A rectangular domain for example is defined like this:
+
+>>> from obspy.clients.fdsn.mass_downloader.domain import RectangularDomain
+>>> domain = RectangularDomain(minlatitude=-10, maxlatitude=10,
+...                            minlongitude=-10, maxlongitude=10)
+
+Additional restrictions like temporal bounds, SEED identifier wildcards,
+and other things are set with the help of
+the :class:`~obspy.clients.fdsn.mass_downloader.restrictions.Restrictions`
+class. Please refer to its documentation for a more detailed explanation of
+the parameters.
+
+>>> from obspy import UTCDateTime
+>>> from obspy.clients.fdsn.mass_downloader import Restrictions
+>>> restrict = Restrictions(
+...     starttime=UTCDateTime(2012, 1, 1),
+...     endtime=UTCDateTime(2012, 1, 1, 1),
+...     network=None, station=None, location=None, channel=None,
+...     reject_channels_with_gaps=True,
+...     minimum_length=0.9,
+...     minimum_interstation_distance_in_m=1000,
+...     channel_priorities=["HH[Z,N,E]", "BH[Z,N,E]"],
+...     location_priorities=["", "00", "01"])
+
+
+Step 2: Storage Options
+~~~~~~~~~~~~~~~~~~~~~~~
+
+After determining what to download, the helpers must know where to store the to
+be downloaded data. This requires some flexibility in case this is integrated
+as a component into a bigger system. An example of this is a toolbox that has a
```

Comment:
There are three "this" in the previous two sentences, and it's a bit unclear what "this" is. It should say "this <something>".